src/dataframe/dataframe.jl

```diff
@@ -263,7 +267,8 @@ end
 function Base.getindex(df::DataFrame, col_inds::AbstractVector)
     selected_columns = index(df)[col_inds]
     new_columns = columns(df)[selected_columns]
-    return DataFrame(new_columns, Index(_names(df)[selected_columns]))
+    new_metadata = metadata(df)[selected_columns]
+    return DataFrame(new_columns, Index(_names(df)[selected_columns]), new_metadata)
```

Comment:
Better use the same pattern as elsewhere and drop the `new_metadata` variable.

Reply:
fixed